lib/resolve.go

```diff
@@ -0,0 +1,92 @@
+package lib
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/qri/base/dsfs"
+	"github.com/qri-io/qri/dsref"
+	"github.com/qri-io/qri/fsi"
+	"github.com/qri-io/qri/resolve"
+)
+
+// assert at compile time that instance is a RefResolver
+var _ resolve.RefResolver = (*Instance)(nil)
+
+// ParseAndResolveRef combines reference parsing and resolution
+func (inst *Instance) ParseAndResolveRef(ctx context.Context, refStr string) (dsref.Ref, error) {
```

Comment:
Is this meant to be exposed to applications that embed qri as a library? Or by `cmd` or `api`? One of the reasons I put the Loader in its own package was so that `lib` could use it, but it wouldn't be accessible by consuming `lib`.

Reply:
we could just expose `Instance.ResolveRef`, but yes I think reference resolution should be exported. It would be very helpful when instrumenting resolution times within a testground test, especially in the p2p context.

Other places where I'd want to see this exported are admittedly esoteric, but pinging the network for a dataset name you expect to exist at some point is another application. basically anything where you're looking to build an index from network calls without moving around any data beyond a reference the way `get` and `log` do.